src/hooks/achievement-rule-checker.js

```
@@ -1,28 +1,53 @@
 // Use this hook to manipulate incoming or outgoing data.
 // For more information on hooks see: http://docs.feathersjs.com/api/hooks.html
 
+const getNestedValue = require('../utils').getNestedValue;
+
 // eslint-disable-next-line no-unused-vars
 module.exports = function (options = {}) {
   return async context => {
 
     const rules = context.app.get('rules');
     const achievementRules = rules['achievements'];
-    // TODO scope
 
     for (const achievementRule of achievementRules) {
       if (await achievementRule.canBeAwarded(context) && await achievementRule.isFulfilled(context)) {
+        const scope = {};
+
+        const hasValidFieldNames = achievementRule.scope.every(scopeFieldName => {
+          if (scopeFieldName !== 'user_id') {
+            const nestedValue = getNestedValue(context.data.context, scopeFieldName);
+            if (nestedValue === undefined) {
+              return false;
+            }
+            scope[scopeFieldName] = nestedValue;
+          }
+          return true;
+        });
+        if (!hasValidFieldNames) {
+          continue;
```

Maybe throw an error here?